### ISSUE TITLE
Server crashes when trying to run server while another server runs

### ISSUE THREAD

**Issue by Wuzzy2** [opensoldat/opensoldat#36]
If you try to run a server but a server already runs on the system, the new server insta-crashes:

```
ctf_Ash by chakapoko maker
Using default weapons mod
[NET DEBUG] Cannot create listen socket.  Failed to bind socket.  Error code 0x00000062.
[NET] Failed to bind to 0.0.0.0:23073
Shutting down server...
Shutting down game networking.
[NET DEBUG] Shutting down low level socket/threading support.
Shutting down server...
An unhandled exception occurred at $00000000004157FD:
EAccessViolation: Access violation
  $00000000004157FD
  $000000000042A9ED
  $000000000040363E

An unhandled exception occurred at $000000000041BF47:
EAccessViolation: Access violation
  $000000000041BF47
  $0000000000418FC1
  $000000000040363E
```

System info:

* OS: Linux 5.7.8 (Arch Linux)
* Pascal: FPC 3.0.4
* libGameNetworkingSockets: built myself, commit 8f60b0e (as recommended here: Soldat/prebuilt-libs#1 (comment))
* libstb: built myself in Soldat repository
* libphysfs: 3.0.2
* make: GNU Make 4.3
* Soldat: commit 078795a

**Comment by zenyd** [comment]
you are trying to connect to the same port where another server is already listening. So I'm not sure this is a bug. Have you tried another port? Like so: `soldatserver -net_port 23074`.

**Comment by Wuzzy2** [comment]
But the server is *crashing*. How is this not a bug? Whether you are supposed to do the thing I did is besides the point, a crash is a crash.

I suggest to catch the error condition (port is already taken) and print out a meaningful error message and quit normally in this case, therefore preventing the crash.

**Comment by helloer** [comment]
I think this bug happens because ShutDown is called twice, first in https://github.com/Soldat/soldat/blob/7780d2948b724970af9f2aaf4fb4e4350d5438d9/server/Server.pas#L1199 and then in https://github.com/Soldat/soldat/blob/7780d2948b724970af9f2aaf4fb4e4350d5438d9/server/Main.pas#L256

Replacing first `ShutDown` with `ProgReady := False;` should fix the issue (_who's looking for good first PR?_  :smiley:), in the long-term we should make `ShutDown` to never cause A/V in such cases.

**Comment by SwamiUmami** [comment]
Should be fixed in #85 
When reviewing code, I saw that Steam builds may also call `ShutDown` twice when an error occurs:
https://github.com/Soldat/soldat/blob/644d796dcea48f7df1b93a72fc558440763378e7/server/Server.pas#L660
I believe changing it to `Exit;` should fix it, but I couldn't make a Steam build to test it.

Apart from that, setting `ProgReady` to False seems like the only way to terminate server's execution, so maybe we could use it more consistently, which would also make it clearer that `ShutDown` should not be called directly to quit.
Currently `ProgReady` is set to True in the middle of `ActivateServer`:
https://github.com/Soldat/soldat/blob/644d796dcea48f7df1b93a72fc558440763378e7/server/Server.pas#L688
Wouldn't it be better to set it to True as soon as server starts, and then we can terminate it by setting `ProgReady` to False? It would then help with consistent quitting when an error occurs: `ProgReady := False;` followed by an optional `Exit;`

In current code, on some failures we [call `ShutDown`](https://github.com/Soldat/soldat/blob/644d796dcea48f7df1b93a72fc558440763378e7/server/Server.pas#L660), in other cases we [set `ProgReady := False` then `Exit`](https://github.com/Soldat/soldat/blob/644d796dcea48f7df1b93a72fc558440763378e7/server/Server.pas#L609), and in some other cases we [raise an exception](https://github.com/Soldat/soldat/blob/644d796dcea48f7df1b93a72fc558440763378e7/server/Server.pas#L747)...
